device-data/linking-events.md

```diff
@@ -1 +1,239 @@
 <!-- non-generated document! all areas editable -->
+
+## Linking Events
+
+Not all events from diabetes management devices in the Tidepool data model(s) are independent. Some events are logically related—perhaps even roughly in a cause-and-effect relationship. In cases where there is a logical connection between events, we endeavor to preserve that connection in the standardized form of the data stored in the Tidepool cloud.
+
+At the time of the drafting of this document in June, 2016, new platform APIs are currently under construction at Tidepool. These will replace the ["jellyfish"](https://github.com/tidepool-org/jellyfish 'Tidepool on GitHub: jellyfish') legacy diabetes data ingestion service, among other services. The strategies for linking events differ slightly when ingesting data through the new platform API versus through the legacy jellyfish API, although *the storage and client data models for linked events are identical, no matter which service was used to ingest the data*. This document covers both ingestion strategies, so pay close attention to which section you're reading!
+
+### List of all (potentially) linked events
+
+- `wizard` -> `bolus` ([details](./types/wizard.md#bolus))
+- `alarm` -> `status` ([details](./types/deviceEvent/alarm.md#status))
+- `reservoirChange` -> `status` ([details](./types/deviceEvent/reservoirChange.md#status))
+
+### Linking events through the legacy jellyfish ingestion API
+
+> DEPRECATED
+
+The legacy jellyfish ingestion service did not use [GUIDs](https://en.wikipedia.org/wiki/Globally_unique_identifier 'Wikipedia: Globally unique identifier') for the `id` field on ingested events. Instead, the `id` field was the *determinstic* result of a hash of several fields, always including `type`, `time`, and `deviceId` plus various other fields depending on the `type` (e.g., including `subType` where relevant). One of the reasons that this deterministic hashing strategy was chosen was because it made interlinking events easy *even when events were not uploaded to the platform in order*.
```

Comment:
Nevermind, the following paragraph makes it clear that it is indeed the case (although swap wizard and bolus in previous comment)